tictactoe/src/components/Board.js

```diff
@@ -1,16 +1,70 @@
-import React from "react";
+import React, { useEffect } from "react";
 import Square from "./Square";
 import "./Board.css";
+import { calculateWinner } from "../helpers/calculateWinner";
 
-export default function Board({ playerMovePositions }) {
-  //console.log(playerMovePositions);
+export default function Board({
+  playerMovePositions,
+  currentPlayer,
+  setCurrentPlayer,
+  setPlayerMovePosition,
+  isOver,
+  setGameOver,
+}) {
+  useEffect(() => {
+    console.log("Phomolo Phiri");
+    const result = checkWinner();
+    if (result) {
+      alert(`${result} has won the game!`);
+      stopGame();
+    }
+  }, [playerMovePositions]);
+  function handleClick(index) {
+    //use a copy of the state to not mutate the state directly.
+    if (!isOver) {
+      const playerMoves = [...playerMovePositions];
+
+      if (currentPlayer === "X") {
+        playerMoves[index] = "X";
+        setCurrentPlayer("O");
+      } else {
+        playerMoves[index] = "O";
+        setCurrentPlayer("X");
+      }
+
+      setPlayerMovePosition(playerMoves);
+    } else {
+      alert("The game is over. Restart it to play again.");
+    }
+  }
+
+  function checkWinner() {
+    let winner = calculateWinner(playerMovePositions);
+    if (winner == "X") {
+      return winner;
+    } else if (winner == "O") {
+      return winner;
```

Comment:
Could benefit from a tenery statement

```
const theWinner = winner === 'X' ? 'X' : 'O'
return theWinner
```

or alternatively

```
return winner === 'X' ? 'X' : 'O'
``` 
to have it as a oneliner

Reply:
Will consider using the new ES6 features when I clean up the code. Nice catch. These old ways are out here making me look like a boomer.